### DISCUSSION TITLE
mimir tsdb issue with flush_blocks_on_shutdown

### DISCUSSION THREAD

**Discussion by kadhamecha-conga** [grafana/mimir#5473] [Help and support]
hi team,

i'm checking the the flush_blocks_on_shutdown to move data from pvc to s3 backend storage, but post killing pod the data is still in pvc , not moving to s3 storage.

version : 
mimir 2.7.1

config 
```
  tsdb:
    dir: /data/tsdb
    flush_blocks_on_shutdown: true
```
before restart 
```
/data/tsdb/anonymous $ du -sh *
86.6M   01H510H9MFW41YMAJNC7V4SZS4
86.5M   01H517D0WFT1WZBBZEJECMNZ1C
86.2M   01H51E8R4FQPMR41NND5449ME9
85.4M   01H51N4FCFSXPMEFRA813ZCKQ0
87.6M   01H51W06MFQ38ETRH54AS05J1A
80.8M   01H522VXWFNRNFHBQR7M8V54M2
75.4M   01H526G684Y8TKQKJHV5J63NQ9
27.7M   01H526GHBNH0PF1VNA0EGY1BCY
33.6M   01H527KC67RKS1QT61W84HMHQY
4.0K    chunks_head
4.0K    mimir.shipper.json
4.0K    thanos.shipper.json
373.0M  wal
```

after restart of pod
```
/data/tsdb/anonymous $ du -sh *
86.6M   01H510H9MFW41YMAJNC7V4SZS4
86.5M   01H517D0WFT1WZBBZEJECMNZ1C
86.2M   01H51E8R4FQPMR41NND5449ME9
85.4M   01H51N4FCFSXPMEFRA813ZCKQ0
87.6M   01H51W06MFQ38ETRH54AS05J1A
80.8M   01H522VXWFNRNFHBQR7M8V54M2
75.4M   01H526G684Y8TKQKJHV5J63NQ9
27.7M   01H526GHBNH0PF1VNA0EGY1BCY
33.6M   01H527KC67RKS1QT61W84HMHQY
29.8M   01H5284S4P3WDBT9XARD9QRX9Q
4.0K    chunks_head
4.0K    mimir.shipper.json
4.0K    thanos.shipper.json
309.7M  wal
```

Please let me know if you require any additional information or if there are any specific logs or debug output that would be helpful for diagnosing the issue.

thanks.

**Comment by 56quarters** [comment]
Flushing to object storage doesn't remove data from ingester disks, it just makes sure it has been uploaded. Ingesters still keep data for 12h and serve queries for data < 12h old. There's some more information here: https://grafana.com/docs/mimir/latest/references/architecture/components/querier/#grafana-mimir-querier

**Reply by kadhamecha-conga** [reply]
hi @56quarters ,

how to verify it has been uploaded to backend, can you please help ?
